smart_control/dataset/dataset.py

```diff
@@ -1,96 +1,469 @@
+    return df
+
+
+class BuildingDatasetPartition:
+  """A helper class for handling a specific dataset partition.
+
+  Args:
+    building_dataset (BuildingDataset): The building dataset.
+    partition_id (str): The identifier of a partition in the specified dataset
+      (e.g. "2022_a").
+
+  Example:
+    ```python
+    ds = BuildingDataset(building_id='sb1', download=True)
+    partition = BuildingDatasetPartition(
+       building_dataset=ds, partition_id='2022_a'
+    )
+    ```
+  """
+
+  def __init__(self, building_dataset: BuildingDataset, partition_id: str):
+    self.ds = building_dataset
+    self.partition_id = partition_id
+
+    if self.partition_id not in self.ds.partition_ids:
+      raise ValueError(f"Invalid partition: {self.partition_id}.")
+
+  @property
+  def partition_dirpath(self):
+    return os.path.join(self.ds.tabular_dirpath, self.ds.building_id, self.partition_id)  # pylint:disable=line-too-long
+
+  @property
+  def data_filepath(self):
+    return os.path.join(self.partition_dirpath, "data.npy.npz")
+
+  @cached_property
+  def data(self) -> np.lib.npyio.NpzFile:
+    """Time-series data for the dataset partition.
+
+    This property returns an `np.lib.npyio.NpzFile` object, which allows
+      dictionary-like access to NumPy arrays stored within a compressed
+      `.npz` archive. The arrays contain time-series data, where the first
+      dimension typically represents the time steps.
+
+    Returns:
+      A dictionary-like numpy object with the following keys:
+
+        - `'observation_value_matrix'`
+        - `'action_value_matrix'`
+        - `'reward_value_matrix'`
+        - `'reward_info_value_matrix'`
+
+    """
+    return np.load(self.data_filepath)
+
+  @property
+  def metadata_filepath(self):
+    return os.path.join(self.partition_dirpath, "metadata.pickle")
+
+  @cached_property
+  def metadata(self) -> dict:
+    """Metadata describing the partition [data](./#smart_control.dataset.dataset.BuildingDatasetPartition.data).
+
+    Returns:
+      A dictionary containing the following keys:
+
+        - `'action_ids'`: dictionary mapping action string identifiers to
+          integer indices, for example: `{'action_id@setpoint': 0}`
+
+        - `'action_timestamps'`: list of sequential timestamps, representing the
+          time of each action
+
+        - `'observation_ids'`: dictionary mapping observation string identifiers
+          to integer indices, for example: `{'obs_id@sensor': 0}`
+
+        - `'observation_timestamps'`: list of sequential of timestamps,
+          representing the time of each observation
+
+        - `'reward_ids'`: dictionary mapping reward string identifiers to
+          integer indices, for example: `{'reward_id@type': 0}`
+
+        - `'reward_timestamps'`: list of sequential timestamps, representing the
+          time of each reward
+
+        - `'reward_info_timestamps'`: list of sequential timestamps, related to
```

Comment:
question: what is the difference between between (`reward_timestamps` / `reward_value_matrix`) and (`reward_info_timestamps` / `reward_info_value_matrix`)? they both use the `reward_ids`. 

@JudahGoldfeder